### PR TITLE
Adds WP_Term_Query support

### PR DESCRIPTION
**Pull request by JasonHoffmann**
Creates a class for terms, and sets default parameters there. The one difference from the other classes is that we need to perform an extra count query in order to get the total count (equivalent of found_posts or get_total, neither of which exist for WP_Term_Query)

### CODE REVIEW

**Comment by zachwills** [comment]
@JasonHoffmann nice work. Can you add unit tests to cover this? I think we have tests for posts and users currently.

File: includes/batches/class-batch-terms.php [hunk 1]
```diff
@@ -0,0 +1,77 @@
+<?php
+/**
+ * Users batch class.
```
Comment (zachwills) [inline includes/batches/class-batch-terms.php]: s/b `Terms`

File: includes/batches/class-batch-terms.php [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+ * @package Locomotive/Batch
+ */
+
+namespace Rkv\Locomotive\Batches;
+
+use WP_Term_Query;
+use Rkv\Locomotive\Abstracts\Batch;
+
+/**
+ * Batch Users class.
```
Comment (zachwills) [inline includes/batches/class-batch-terms.php]: s/b `Terms`

File: includes/batches/class-batch-terms.php [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+	 * Get results function for the registered batch process.
+	 *
+	 * @return array \WP_Term_Query->get_results() result.
+	 */
+	public function batch_get_results() {
+		$query = new WP_Term_Query( $this->args );
+
+		// Need to do a count query in order to get all possible terms as an integer.
+		$count_args = wp_parse_args( array( 'fields' => 'count', 'offset' => 0 ), $this->args );
+		$count = new WP_Term_query( $count_args );
```
Comment (jjeaton) [inline includes/batches/class-batch-terms.php]: s/b WP_Term_Query.

**Comment by JasonHoffmann** [comment]
Issues fixed

**Pull request merged**